mdn/src/main/java/org/apache/james/mdn/MDN.java

```diff
@@ -233,6 +238,17 @@ public BodyPart computeReportPart() throws MessagingException {
         return mdnPart;
     }
 
+    public BodyPart computeOriginalMessagePart() throws MessagingException {
+        Preconditions.checkState(message.isPresent());
+        MimeBodyPart originalMessagePart = new MimeBodyPart();
+        try {
+            originalMessagePart.setText(new String(DefaultMessageWriter.asBytes(message.get()), StandardCharsets.UTF_8));
```

Comment:
I would prefer:

```suggestion
            originalMessagePart.setContent(new String(DefaultMessageWriter.asBytes(message.get()), StandardCharsets.UTF_8), "message/rfc822");
```

Do we have unit tests for this method?